pkg/agent/inteface.go

```diff
@@ -76,6 +76,19 @@ type AgentAddonOptions struct {
 	// SupportedConfigGVRs is a list of addon supported configuration GroupVersionResource
 	// each configuration GroupVersionResource should be unique
 	SupportedConfigGVRs []schema.GroupVersionResource
+
+	// AgentDeployClusterInformerFilter defines the filter function for the cluster informer when deploying the addon
+	// agent. Addons that need information from the ManagedCluster resource when deploying the agent should use this
+	// field to set what information they need, otherwise the expected/up-to-date agent may be deployed delayed since
+	// the default filter func returns false when the ManagedCluster resource is updated.
+	//
+	// For example, the agentAddon needs information from the ManagedCluster annotation, it can set the filter function
+	// like:
+	//
+	//	AgentDeployClusterInformerFilter: func(old, new *clusterv1.ManagedCluster) bool {
+	//	 return !equality.Semantic.DeepEqual(old.Annotations, new.Annotations)
+	//	}
+	AgentDeployClusterInformerFilter func(old, new *clusterv1.ManagedCluster) bool
```

Comment:
```suggestion
	AgentDeployTriggerClusterFilter func(old, new *clusterv1.ManagedCluster) bool
```
something like this?

Reply:
ok, fixed